app/javascript/app/components/modal-overview/modal-overview-component.jsx

```diff
@@ -0,0 +1,95 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import Modal from 'components/modal/modal-component';
+import { NavLink } from 'react-router-dom';
+import ModalHeader from 'components/modal/modal-header-component';
+import NoContent from 'components/no-content';
+import cx from 'classnames';
+import startCase from 'lodash/startCase';
+import isArray from 'lodash/isArray';
+import styles from './modal-overview-styles.scss';
+
+const MetadataProp = ({ title, children }) => (
+  <p className={styles.text}>
+    <span className={styles.textHighlight}>{startCase(title)}: </span>
+    {children}
+  </p>
+);
+
+MetadataProp.propTypes = {
+  title: PropTypes.string,
+  children: PropTypes.node
+};
+
+class ModalOverview extends PureComponent {
+  constructor() {
+    super();
+    this.handleOnRequestClose = this.handleOnRequestClose.bind(this);
+    this.state = {
+      selectedIndex: 0
+    };
+  }
+
+  handleOnRequestClose() {
+    this.props.onRequestClose();
+  }
+
+  renderData() {
```

Comment:
This whole function seems rather elaborate. Can you the data not be better mapped to render simply. That is why we have selectors in the first place, to keep the components stupid and reusable. This seem like a complicated specific case.